container/lxd/manager_test.go

```diff
@@ -119,31 +126,25 @@ func prepNetworkConfig() *container.NetworkConfig {
 }
 
 func (s *managerSuite) TestContainerCreateDestroy(c *gc.C) {
-	ctrl := gomock.NewController(c)
+	ctrl := s.setup(c)
 	defer ctrl.Finish()
-	cSvr := s.NewMockServer(ctrl)
-	s.patch(cSvr)
+	s.patch()
+	s.makeManager(c)
 
-	manager := s.makeManager(c, cSvr)
 	iCfg := prepInstanceConfig(c)
-	hostName, err := manager.Namespace().Hostname(iCfg.MachineId)
+	hostName, err := s.manager.Namespace().Hostname(iCfg.MachineId)
 	c.Assert(err, jc.ErrorIsNil)
 
 	// Operation arrangements.
-	startOp := lxdtesting.NewMockOperation(ctrl)
-	startOp.EXPECT().Wait().Return(nil)
-
-	stopOp := lxdtesting.NewMockOperation(ctrl)
-	stopOp.EXPECT().Wait().Return(nil)
+	s.expectStartOp(ctrl)
+	s.expectStopOp(ctrl)
+	s.expectDeleteOp(ctrl)
 
-	deleteOp := lxdtesting.NewMockOperation(ctrl)
-	deleteOp.EXPECT().Wait().Return(nil)
-
-	exp := cSvr.EXPECT()
+	exp := s.cSvr.EXPECT()
 
 	// Arrangements for the container creation.
-	expectCreateContainer(ctrl, cSvr, "juju/xenial/"+s.Arch(), "foo-target")
-	exp.UpdateContainerState(hostName, lxdapi.ContainerStatePut{Action: "start", Timeout: -1}, "").Return(startOp, nil)
+	s.expectCreateContainer(ctrl)
```

Comment:
some noise in this file, i wanted to update to the pattern of expect*() for setup while here.